src/main/java/org/truffleruby/core/rope/Rope.java

```diff
@@ -161,8 +158,15 @@ public byte get(int index) {
         return getByteSlow(index);
     }
 
+    @Override
+    public String toString() {
+        // This is designed to not have any side effects - compare to getString
+        return RopeOperations.decode(encoding, RopeOperations.flattenBytes(this));
+    }
+
     /** Should only be used by the parser */
-    public final String getString() {
+    public final String normaliseAndGetJavaString() {
```

Comment:
What does the normalize mean here?
How about simply `getJavaString()`?

Reply:
Normalise as in to get it into a normal form. I suppose I could call it `flattenAndGetJavaString`. I didn't want to call it `getJavaString` as it has a significant side effect, which is I think also partially what it's being used for.

Reply:
There is also unicode normalization in the context of strings so `normalise` seems confusing.
There are many methods that implicitly flatten, yet I feel the flatten is part of the implementation details, and is necessary for avoiding performance issues when repetitively calling the method on the same Rope/RubyString.
Notably `RubyStringLibrary#getJavaString` is very similar (but works in PE code).

Could you rename it to `getJavaString()` for consistency with RubyStringLibrary?